src/resolution_functions/instrument.py

```diff
@@ -630,40 +674,124 @@ def available_models(self) -> list[str]:
+        :term:`instrument`.
 
-        All :term:`models<model>` available for this :term:`version` of this :term:`instrument`,
-        and all their :term:`configurations<configuration>` are listed.
+        Only includes the versioned models, i.e. lists all versions of all models, but does not list
+        the recommended models.
 
         Returns
         -------
-        models_and_configurations
-            All models and all their configurations.
+        available_models
+            A list of all available unique models.
         """
-        return {model_name: list(model['configurations'].keys())
-                for model_name, model in self._models.items()}
+        return [name for name, value in self._models.items() if not isinstance(value, str)]
 
-    @property
-    def all_available_models_options(self) -> dict[str, dict[str, list[str]]]:
+    def format_available_models_and_configurations(self) -> str:
         """
-        A dictionary mapping each available :term:`model`, to the user
-        :term:`configurations<configuration>` and their :term:`options<option>`.
+        Formats all available :term:`models<model>` and :term:`configurations<configuration>` into a
+        table.
 
-        All :term:`models<model>` available for this :term:`version` of this :term:`instrument`, all
-        the :term:`configurations<configuration>` of each of the :term:`models<model>`, and all the
-        :term:`options<option>` for each of the :term:`configurations<configuration>`, are listed.
+        The table shows each :term:`model` and either which other :term:`model` it is an alias for,
+        or all the :term:`configurations<configuration>` required by that :term:`model`.
 
         Returns
         -------
-        everything
-            All models, all their configurations, and all their options.
+        str
+            A string containing the nicely formatted table.
+        """
+        contents = [['MODEL', 'ALIAS FOR', 'CONFIGURATIONS']]
+        longest_name, longest_alias, longest_config = 5, 9, 14
+
+        for model_name, model_data in self._models.items():
+            length = len(model_name)
+            if length > longest_name:
+                longest_name = length
+
+            if isinstance(model_data, str):
+                contents.append([model_name, model_data, None])
+
+                length = len(model_data)
+                if length > longest_alias:
+                    longest_alias = length
+            else:
+                for i, config_name in enumerate(model_data['configurations']):
+                    if i == 0:
+                        contents.append([model_name, None, config_name])
+                    else:
+                        contents.append([None, None, config_name])
+
+                    length = len(config_name)
+                    if length > longest_config:
+                        longest_config = length
```

Comment:
While it requires 2 runs through the elements, I would honestly not compute the length in this building-the-list stage as it makes the loop a lot more complex. 

I would iterate through the compiled list and compute the longest in each column after this. The list will be short (<1k elements), so the cost of multiple runs (unless this function is called a lot) is irrelevant. 

Pulling the logic out and using:
```
for name, alias, config in contents:
    longest_name  = max(longest_name, len(name))
    longest_alias  = max(longest_alias, len(alias))
    longest_config  = max(longest_config, len(config))
```
(or something more efficient) at the end makes the purpose clear.

Reply:
Maybe inside `_format_table`, we could do (though we would either need to do masking or change the `None` into e.g. `"-"`):

```
contents = np.array(contents)
longest = np.max(np.vectorize(len)(b), axis=0)
```

Reply:
Seems promising. It should be done after the None->str conversion, just in case we decide to replace None with `"----------"` 😉

Reply:
I think it'd be better to do that upstream, in `format_available_models_and_configurations` and `format_available_models_options`, and completely eliminite the `None`s as well as the pre-processing.

Reply:
With those sort of details already taken care of, it becomes more practical to have a generic table formatter that e.g. supports multiple criteria of how blocks are split. (Or maybe just requires the blocks to be pre-split as well.)